File: c/factorial.c
#include <stdio.h>
 
int main()
{
  int i, n, fact = 1;
 
  printf("Enter a number to calculate its factorial\n");
  scanf("%d", &n);

  for (i = 1; i <= n; i++){
      fact = fact * i;
  }
 
  printf("Factorial of %d = %d\n", n, fact);
 
  return 0;
}

File: GO/factorial.go
package main

import "fmt"

func main() {

	for a := 1; a <= 10; a++ {
		fmt.Println(a," factorial is", factorial(a))
	}
}

func factorial(number int) int {
	var result int = 1

	for i := 1; i <= number; i++ {
		result = result * i
	}

	return result
}
